src/NuGet.Indexing/RankingsScoreQuery.cs

```diff
@@ -39,41 +38,54 @@ public override string ToString()
         private class RankingScoreProvider : CustomScoreProvider
         {
             private readonly double _baseBoost;
-            private readonly IDictionary<string, int> _rankings;
-            private readonly string[] _ids;
+            private readonly RankingsHandler.RankingResult _rankings;
+            private readonly string _readerName;
 
-            public RankingScoreProvider(IndexReader reader, IDictionary<string, int> rankings, double baseBoost)
+            public RankingScoreProvider(IndexReader reader, RankingsHandler.RankingResult rankings, double baseBoost)
                 : base(reader)
             {
                 _rankings = rankings;
-                _ids = FieldCache_Fields.DEFAULT.GetStrings(reader, "Id");
                 _baseBoost = baseBoost;
+
+                SegmentReader segmentReader = reader as SegmentReader;
+
+                _readerName = segmentReader != null
+                    ? segmentReader.SegmentName
+                    : string.Empty;
```

Comment:
And in general, can you add some documentation to this file..

---
In reply to: [59756131](https://github.com/NuGet/NuGet.Services.Metadata/pull/64#discussion_r59756131) [](ancestors = 59756131)